scala/lambdas/ingest-parsed-court-document-event-handler/src/main/scala/uk/gov/nationalarchives/ingestparsedcourtdocumenteventhandler/FileProcessor.scala

```diff
@@ -174,22 +173,17 @@ class FileProcessor(
           .flatMap { stream =>
             if (!tarEntry.isDirectory) {
               val id = uuidGenerator()
-              val checksumResult = stream
-                .through(fs2.hashing.Hashing[IO].hash(HashAlgorithm.SHA256))
-                .flatMap(hash => Stream.emits(hash.bytes.toList))
-                .through(fs2.text.hex.encode)
-                .compile
-                .to(string)
               Stream.eval[IO, (String, FileInfo)](
-                stream.chunks
-                  .map(_.toByteBuffer)
-                  .toPublisherResource
-                  .use(pub => s3.upload(uploadBucket, id.toString, FlowAdapters.toPublisher(pub)))
-                  .flatMap { res =>
-                    checksumResult.map { checksum =>
+                stream.compile.toList.flatMap { bytes =>
+                  Stream
+                    .emit[IO, ByteBuffer](ByteBuffer.wrap(bytes.toArray))
+                    .toPublisherResource
+                    .use(pub => s3.upload(uploadBucket, id.toString, FlowAdapters.toPublisher(pub)))
+                    .map { _ =>
+                      val checksum = DigestUtils.sha256Hex(bytes.toArray)
                       tarEntry.getName -> FileInfo(id, tarEntry.getSize, tarEntry.getName.split('/').last, checksum, URI.create(s"s3://$uploadBucket/${id.toString}"))
```

Comment:
As we have now started supporting multiple checksums elsewhere, it might be useful to modify the `FileInfo` case class within this to call the checksum as `sha256Checksum` so we know, in this type of ingest we are generating only one checksum and it is always `SHA256` 

`case class FileInfo(id: UUID, fileSize: Long, fileName: String, sha256Checksum: String, location: URI)`

Reply:
Good point. I've changed it in the case class and changed the variable name on this line.